### ISSUE TITLE
Parallel usage results in a "socket hang up" error. systemPort have set but still have the problem.

### ISSUE THREAD

**Issue by Underson888** [appium/python-client#903]
Here's my code:

python

from appium import webdriver
from appium.webdriver.common.touch_action import TouchAction
import threading

# Device 1's configuration
desired_caps1 = {
    "appium:appActivity": ".Settings",
    "appium:appPackage": "com.android.settings",
    "appium:automationName": "uiautomator2",
    "appium:udid": "xxxx",
    "platformName": "Android",
    "appium:systemPort": 8310
}

# Device 2's configuration
desired_caps2 = {
    "appium:appActivity": ".Settings",
    "appium:appPackage": "com.android.settings",
    "appium:automationName": "uiautomator2",
    "appium:udid": "192.168.137.58:5555",
    "platformName": "Android",
    "appium:systemPort": 8322
}

def run_test_on_device(desired_caps):
    driver = webdriver.Remote('http://127.0.0.1:4723/wd/hub', desired_caps)  # Corrected URL

    width = driver.get_window_size()["width"]
    height = driver.get_window_size()["height"]

    action = TouchAction(driver)
    action.tap(x=width // 2, y=height // 2).perform()

    driver.quit()

# Use threading for parallel execution
thread1 = threading.Thread(target=run_test_on_device, args=(desired_caps1,))
thread2 = threading.Thread(target=run_test_on_device, args=(desired_caps2,))

thread1.start()
thread2.start()

thread1.join()
thread2.join()

I have set the different **systemPort** for each devices, but I still have the problem.

How can I sove that, I'm very need to be solved, thanks.

**Comment by KazuCocoa** [comment]
Could you share the appium log as well? (as GIST)

**Comment by Underson888** [comment]
> Could you share the appium log as well? (as GIST)

Ok,bro. This is the link:https://gist.github.com/Underson888/8871bca07b3241d7aa6f810e9a7bc16c

**Comment by KazuCocoa** [comment]
Did they work with a single thread? The port config itself looks working

**Comment by Underson888** [comment]
> Did they work with a single thread? The port config itself looks working

Yes,each device is controlled by a single thread,just like the code mentioned.

**Comment by Underson888** [comment]
> Did they work with a single thread? The port config itself looks working

thread1 = threading.Thread(target=run_test_on_device, args=(desired_caps1,))
thread2 = threading.Thread(target=run_test_on_device, args=(desired_caps2,))

thread1.start()
thread2.start()

thread1.join()
thread2.join()

Is it correct for the driver?
